dd-trace-core/src/main/java/datadog/trace/lambda/NoOpAdapter.java

```diff
@@ -0,0 +1,49 @@
+package datadog.trace.lambda;
+
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.JsonReader;
+import com.squareup.moshi.JsonWriter;
+import com.squareup.moshi.Moshi;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Set;
+
+public final class NoOpAdapter<T> extends JsonAdapter<T> {
+  private final JsonAdapter<T> delegate;
+  private final String typeToSkip;
+
+  private NoOpAdapter(JsonAdapter<T> delegate, String typeToSkip) {
+    this.delegate = delegate;
+    this.typeToSkip = typeToSkip;
+  }
+
+  @Override
+  public T fromJson(JsonReader reader) throws IOException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void toJson(JsonWriter writer, T value) throws IOException {
+    if (null != value && value.getClass().getName().equals(typeToSkip)) {
+      writer.beginObject();
+      writer.endObject();
+      return;
+    }
+    delegate.toJson(writer, value);
+  }
+
+  public static <T> Factory newFactory(final String type) {
+    return new Factory() {
+      @Override
+      public JsonAdapter<?> create(
+          Type requestedType, Set<? extends Annotation> annotations, Moshi moshi) {
+        if (null != requestedType && requestedType.toString().endsWith(type)) {
```

Comment:
If you wanted to match the check in the adapter then you could use:
```suggestion
        if (requestedType instanceof Class<?>
            && ((Class<?>) requestedType).getClass().getName().equals(typeToSkip)) {
```
which should be ok as the two current types are not generic, and so their reflected `Type` here would be `Class`.